src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewViewModel.cs

```diff
@@ -0,0 +1,846 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using TreeNodeSelectionState = Windows.UI.Xaml.Controls.TreeViewNode.TreeNodeSelectionState;
+
+namespace Windows.UI.Xaml.Controls
+{
+	internal partial class TreeViewViewModel : ObservableVector<object>
+	{
+		private bool m_isContentMode;
+		private SelectedTreeNodeVector m_selectedNodes;
+		private SelectedItemsVector m_selectedItems;
+		private Dictionary<object, TreeViewNode> m_itemToNodeMap;
+
+		private TreeViewNode m_originNode;
+		private TreeViewList m_TreeViewList;
+
+		public TreeViewViewModel()
+		{
+			var selectedNodes = new SelectedTreeNodeVector();
+			selectedNodes.SetViewModel(this);
+			m_selectedNodes = selectedNodes;
+
+			var selectedItems = new SelectedItemsVector();
+			selectedItems.SetViewModel(this);
+			m_selectedItems = selectedItems;
+
+			m_itemToNodeMap = new Dictionary<object, TreeViewNode>();
+		}
+
+		//ViewModel.~ViewModel()
+		//{
+		//	if (m_rootNodeChildrenChangedEventToken.value != 0)
+		//	{
+		//		if (var origin = m_originNode.safe_get())
+		//        {
+		//			get_self<TreeViewNode>(origin).ChildrenChanged(m_rootNodeChildrenChangedEventToken);
+		//		}
+
+		//		ClearEventTokenVectors();
+		//	}
+		//}
+
+		internal void ExpandNode(TreeViewNode value)
+		{
+			value.IsExpanded = true;
+		}
+
+		internal void CollapseNode(TreeViewNode value)
+		{
+			value.IsExpanded = false;
+		}
+
+		internal event TypedEventHandler<TreeViewNode, object> NodeExpanding;
+
+		internal event TypedEventHandler<TreeViewNode, object> NodeCollapsed;
+
+		internal void SelectAll()
+		{
+			UpdateSelection(m_originNode, TreeNodeSelectionState.Selected);
+		}
+
+		internal void ModifySelectByIndex(int index, TreeNodeSelectionState state)
+		{
+			var targetNode = GetNodeAt(index);
+			UpdateSelection(targetNode, state);
+		}
+
+		internal int Size
+		{
+			get
+			{
+				var inner = GetVectorInnerImpl();
+				return inner.Count;
+			}
+		}
+
+		private object GetAt(int index)
+		{
+			TreeViewNode node = GetNodeAt(index);
+			return IsContentMode ? node.Content : node;
+		}
+
+		private new int IndexOf(object value)
+		{
+			var inner = GetVectorInnerImpl();
+			return inner.IndexOf(value);
+		}
+
+		private object[] GetMany(int startIndex)
+		{
+			var inner = GetVectorInnerImpl();
+			if (IsContentMode)
+			{
+				var vector = new List<object>();
+				int size = Size;
+				for (int i = 0; i < size; i++)
+				{
+					vector.Append(GetNodeAt(i).Content);
+				}
+				return vector.Skip(startIndex).ToArray();
+			}
+			return inner.Skip(startIndex).ToArray();
+		}
+
+		internal TreeViewNode GetNodeAt(int index)
+		{
+			var inner = GetVectorInnerImpl();
+			return (TreeViewNode)inner[index];
+		}
+
+		//void ViewModel.SetAt(uint32_t index, IInspectable const& value)
+		//{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)